storage/src/desktop/storage_reference_desktop.cc

```diff
@@ -228,6 +230,7 @@ Future<void> StorageReferenceInternal::DeleteLastResult() {
 }
 
 // Handy utility function, since REST calls have similar setup and teardown.
+// Can potentially block getting Future results for the Request.
```

Comment:
This change does introduce this potential blocking.
For async storage operations, is PrepareRequest already called on a separate thread? I feel like we just need to be sure that this will not cause async operations to block the main thread.

Reply:
I renamed it to ...Blocking, to make it a bit more obvious. All the calls are currently done within a BlockingResponse, which already had a blocking call for the rest call, so I believe it should be fine.

Reply:
Sounds good. Yea that should be enough to ensure that this code doesn't accidentally block the main thread.